Src/LogData.c

```diff
@@ -184,8 +195,14 @@ void initializeLogEntry()
     theLogEntry.antennaAltitude = -1;
     theLogEntry.geoidAltitude = -1;
     theLogEntry.altitude = -1;
+    //[1] Using memcpy: does it only update one element of the struct?
+    //[2] when using memcpy to update entire struct, would we need a temp stuct?
+
+    //because struct memory is contiguous, is this correct:
+    //char initValue = 0xFF;
+    //memcpy (&LogEntry, &initValue, sizeof(LogEntry));
```

Comment:
Whoops, I meant `memset(...)`, not `memcpy(...)`. The difference is that `memset(...)` *sets* a certain number of bytes at a pointer to a distinct value, and `memcpy(...)` *copies* a certain number of bytes from one pointer to another.

Right now, what would happen is that it would copy `sizeof(LogEntry)` number of bytes from the memory address of `initValue` to the memory address of `theLogEntry` (Careful of the typo there, you want to copy to the variable, not the type). The issue is that `initValue` is a `uint8_t`, i.e. only 1 byte, and a `LogEntry` variable is 90+ bytes. `memcpy(...)` will not "repeat" the value pointed to in the second argument, but will instead keep reading memory after where `initValue` is defined, which could include other variables, uninitialized garbage data, or areas of memory that aren't allowed to be accessed, which would throw a `Segmentation Fault`, causing the entire system to crash.

Here's the *right* reference page lol: http://www.cplusplus.com/reference/cstring/memset/